src/runtime-rs/src/shim/src/run.rs

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) 2021 Alibaba Cloud
+// Copyright (c) 2021 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::os::unix::fs::OpenOptionsExt;
+use std::os::unix::io::{FromRawFd, RawFd};
+use std::path::Path;
+use std::sync::mpsc::{channel, Receiver, Sender};
+use std::sync::{Arc, Mutex};
+
+use log::error;
+use slog::Drain;
+use ttrpc::server::Server;
+
+use agent::Agent;
+use shim_proto::shim_ttrpc;
+use virtcontainers::{Sandbox, TomlConfig};
+
+use crate::config::load_configuration;
+use crate::event_publisher::EventPublisher;
+use crate::task_service::TaskService;
+use crate::{Error, Result, ShimExecutor, ENV_SOCKET_FD};
+
+impl ShimExecutor {
+    pub fn run(&mut self) {
+        if let Err(e) = self.args.validate(false) {
+            eprintln!("run shim err: {}", e);
```

Comment:
Is this needed or is it just debug that can be deleted? If we need it, Is there no logger we can use at this stage? But why can't the function return a `Result` instead?

Reply:
This `run` method is used as the most outside function to handle "normal run" condition,  so it does not return errors but directly write errors from inner function calls to stderr.  The `start` and `delete` method follows the same logic